src/components/Encounter/EncounterActions.tsx

```diff
@@ -155,41 +157,43 @@ export default function EncounterActions({
             {t("discharge_summary")}
           </Link>
         </Button>
-        <AlertDialog>
-          <AlertDialogTrigger asChild>
-            <Button
-              variant={variant}
-              size={size}
-              className="w-full justify-start"
-            >
-              {t("mark_as_complete")}
-            </Button>
-          </AlertDialogTrigger>
-          <AlertDialogContent>
-            <AlertDialogHeader>
-              <AlertDialogTitle>{t("mark_as_complete")}</AlertDialogTitle>
-              <AlertDialogDescription>
-                {t("mark_encounter_as_complete_confirmation")}
-              </AlertDialogDescription>
-            </AlertDialogHeader>
-
-            <PLUGIN_Component
-              __name="PatientInfoCardMarkAsComplete"
-              encounter={encounter}
-            />
-
-            <AlertDialogFooter>
-              <AlertDialogCancel>{t("cancel")}</AlertDialogCancel>
-              <AlertDialogAction
-                className={cn(buttonVariants({ variant: "primary" }))}
-                onClick={handleMarkAsComplete}
-                data-cy="mark-encounter-as-complete"
+        {!completedEncounterStatus.includes(encounter.status) && (
+          <AlertDialog>
+            <AlertDialogTrigger asChild>
+              <Button
+                variant={variant}
+                size={size}
+                className="w-full justify-start"
               >
                 {t("mark_as_complete")}
-              </AlertDialogAction>
-            </AlertDialogFooter>
-          </AlertDialogContent>
-        </AlertDialog>
+              </Button>
+            </AlertDialogTrigger>
+            <AlertDialogContent>
+              <AlertDialogHeader>
+                <AlertDialogTitle>{t("mark_as_complete")}</AlertDialogTitle>
+                <AlertDialogDescription>
+                  {t("mark_encounter_as_complete_confirmation")}
+                </AlertDialogDescription>
+              </AlertDialogHeader>
+
+              <PLUGIN_Component
+                __name="PatientInfoCardMarkAsComplete"
+                encounter={encounter}
+              />
+
+              <AlertDialogFooter>
+                <AlertDialogCancel>{t("cancel")}</AlertDialogCancel>
+                <AlertDialogAction
+                  className={cn(buttonVariants({ variant: "primary" }))}
```

Comment:
isn't this sufficient? 🤔 

```suggestion
                  className={buttonVariants({ variant: "primary" })}
```

Reply:
It is everywhere like this 😅 but yeah removed it ✅ 
Great, for noticing